### PR TITLE
feat: get_channel_options from dict to tuple

### LINKED ISSUE #482: Change grpc options (channel_arguments) from dict to tuple
**Is your feature request related to a problem? Please describe.**

When changing between librarys we should not expect users to define the channel options differently than the official grpc package itself. 

**Describe the solution you'd like**

We should rewrite that to just use tuples as grpc itself does.

e.g. Link to the Documentation suggests tuple[tuple[...]] `(('grpc.so_reuseport', 0),)` but we expect Dict[str, Any].
https://github.com/camunda-community-hub/pyzeebe/blob/de788a894741bbc8707dc5ca683b5db273ed8407/pyzeebe/channel/camunda_cloud_channel.py#L30

**Additional context**

https://grpc.github.io/grpc/python/glossary.html

I think it (changing options from dict to tuple) should be in another PR.

_Originally posted by @dimastbk in https://github.com/camunda-community-hub/pyzeebe/pull/468#discussion_r1761654389_

### PR DESCRIPTION
**Pull request by felicijus**
Change grpc options (channel_arguments) from dict to tuple

## Changes

- get_channel_options type from `dict[str, Any]` -> `ChannelArgumentType = Sequence[Tuple[str, Any]]`
- dependend channel creation methods

### New Features *(required)*

The function for defining default grpc [channel_arguments](https://grpc.github.io/grpc/python/glossary.html)

### Enhancements *(optional)*

When changing between librarys users can define the channel options like in the official grpc package itself.
See https://grpc.github.io/grpc/python/glossary.html.

## Checklist

- [x] Unit tests
- [x] Documentation

## References

https://grpc.github.io/grpc/python/glossary.html
https://github.com/grpc/grpc/blob/v1.64.x/include/grpc/impl/channel_arg_names.h

Resolves #482

### CODE REVIEW

File: pyzeebe/credentials/typing.py [hunk 1]
```diff
@@ -1,4 +1,7 @@
-from typing import Protocol, Tuple, Union
+from typing import Any, Protocol, Sequence, Tuple, Union
+
+ChannelArgumentType = Sequence[Tuple[str, Any]]
+# from grpc.aio._typing import ChannelArgumentType  # type: ignore[import-untyped]
```
Comment (dimastbk) [inline pyzeebe/credentials/typing.py:4]:
```suggestion
```

File: pyzeebe/credentials/typing.py [hunk 2]
```diff
@@ -1,4 +1,7 @@
-from typing import Protocol, Tuple, Union
+from typing import Any, Protocol, Sequence, Tuple, Union
+
+ChannelArgumentType = Sequence[Tuple[str, Any]]
```
Comment (dimastbk) [inline pyzeebe/credentials/typing.py:3]: Please, move it to https://github.com/camunda-community-hub/pyzeebe/blob/master/pyzeebe/types.py

**Review by dimastbk — APPROVED** [review]

File: pyzeebe/credentials/typing.py (hunk 2 above)
Comment (felicijus) [reply pyzeebe/credentials/typing.py:3]: I will do that and update the documentation, then this should be ready to merge.

**Comment by felicijus** [comment]
I found something that could be a Problem.

`"c:\Users\A113927430\github\pyzeebe\pyzeebe\types.py" is overriding the stdlib module "types"Pylance[reportShadowedImports](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportShadowedImports)`

**Pull request merged**